sops.go

```diff
@@ -318,142 +320,132 @@ type Store interface {
 // MasterKeyCount returns the number of master keys available
 func (m *Metadata) MasterKeyCount() int {
 	count := 0
-	for _, ks := range m.KeySources {
-		count += len(ks.Keys)
+	for _, group := range m.KeyGroups {
+		count += len(group)
 	}
 	return count
 }
 
-// RemoveMasterKeys removes all of the provided keys from the metadata's KeySources, if they exist there.
-func (m *Metadata) RemoveMasterKeys(keys []MasterKey) {
-	for j, ks := range m.KeySources {
-		var newKeys []MasterKey
-		for _, k := range ks.Keys {
-			matchFound := false
-			for _, keyToRemove := range keys {
-				if k.ToString() == keyToRemove.ToString() {
-					matchFound = true
-					break
-				}
-			}
-			if !matchFound {
-				newKeys = append(newKeys, k)
-			}
+func (m *Metadata) UpdateMasterKeysWithKeyServices(dataKey []byte, svcs []keyservice.KeyServiceClient) (errs []error) {
+	if len(svcs) == 0 {
+		return []error{
+			fmt.Errorf("No key services provided, cansnot update master keys."),
 		}
-		m.KeySources[j].Keys = newKeys
 	}
-}
-
-// UpdateMasterKeysIfNeeded encrypts the data key with all master keys if it's needed
-func (m *Metadata) UpdateMasterKeysIfNeeded(dataKey []byte) (errs []error) {
-	for _, ks := range m.KeySources {
-		for _, k := range ks.Keys {
-			err := k.EncryptIfNeeded(dataKey)
-			if err != nil {
-				errs = append(errs, fmt.Errorf("Failed to encrypt new data key with master key %q: %v\n", k.ToString(), err))
+	var parts [][]byte
+	if len(m.KeyGroups) == 1 {
+		// If there's only one key group, we can't do Shamir. All keys
+		// in the group encrypt the whole data key.
+		parts = append(parts, dataKey)
+	} else {
+		var err error
+		if m.ShamirQuorum == 0 {
+			m.ShamirQuorum = len(m.KeyGroups)
+		}
+		log.Printf("Multiple KeyGroups found, proceeding with Shamir with quorum %d", m.ShamirQuorum)
+		parts, err = shamir.Split(dataKey, len(m.KeyGroups), int(m.ShamirQuorum))
```

Comment:
shouldn't this be an int already?

Reply:
Yup. I ran [unconvert](https://github.com/mdempsky/unconvert) to make sure there aren't any other cases of this, and this is the only one in the codebase.